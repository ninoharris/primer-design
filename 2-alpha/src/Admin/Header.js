import React from 'react'
import { Link, withRouter } from 'react-router-dom'


const Header = ({ title, children, location, match, ...rest }) => {
  console.log('Rest of header location info:', rest)
  return (
    <div className="row">
      <div className="Nav">
        <div className="Logo">
          <h4>{title}</h4>
        </div>
        <div>
          {children}
          <Link to="/admin/create"><button className="btn btn-success mr-3">Add new exercise</button></Link>
          <button className="btn btn-info mr-3" onClick={() => { }}>View students</button>
          {location.pathname !== '/admin' ? <Link to="/admin" className="btn btn-success">Back to home</Link> : ''}
          <button className="btn btn-warning mr-3" onClick={() => { }}>Log out</button>
        </div>
      </div>
    </div>
  )
}


export default withRouter(Header)