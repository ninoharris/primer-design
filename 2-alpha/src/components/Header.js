import React from 'react'

const Header = ({
  loggedIn = false,
  username = '',
  completedCount = 0,
}) => {
  const userDisplay = loggedIn ? (
    <div className="logged-in">
      <p>Logged in as: <span className="username">{username}</span></p>
      <p className="exercises-completed">{completedCount} completed</p>
      <button onClick={() => { }} className="btn btn-warning log-out">
        Log Out
      </button>
    </div>
    ) : (
    <div className="logged-out">
      <p>Not currently signed in</p>
      <button onClick={() => { }} className="btn btn-success log-in">
          Log Out
      </button>
    </div>
  )

  return (
    <div className="row">
      <div className="Nav">
        <div className="Logo">
          <h3 className="mb-0">Primer Designer</h3>
        </div>
        <div>
          <button onClick={() => { }} className="btn btn-info" /* TODO: update with tutorial pages */>
            See tutorial
            </button>
          {userDisplay}
        </div>
      </div>
    </div>
  )
}

export default Header