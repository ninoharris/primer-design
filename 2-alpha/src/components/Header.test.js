import React from 'react'
import { shallow } from 'enzyme'
import { Header } from './Header'

test('should render correctly', () => {
  const wrapper = shallow(<Header/>)
  expect(wrapper).toMatchSnapshot()
})

test('Header, without props, should display a logged-out container with a log in button', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper.find('.logged-out').length).toBe(1)
  expect(wrapper.find('.logged-in').length).toBe(0)
})

describe('Header, when logged in', () => {
  const username = 'sedm4648'
  const wrapper = shallow(<Header loggedIn={true} username={username} />)
  test('should display a logged in section with log out button', () => {
    expect(wrapper.find('.logged-in').length).toBe(1)
    expect(wrapper.find('.log-out').length).toBe(1)
  })
  test('should display the username', () => {
    expect(wrapper.find('.username').text()).toBe(username)
  })
})