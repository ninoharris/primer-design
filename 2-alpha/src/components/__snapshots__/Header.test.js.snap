// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="row"
>
  <div
    className="Nav"
  >
    <div
      className="Logo"
    >
      <h3
        className="mb-0"
      >
        Primer Designer
      </h3>
    </div>
    <div>
      <button
        className="btn btn-info"
        onClick={[Function]}
      >
        See tutorial
      </button>
      <div
        className="logged-out"
      >
        <p>
          Not currently signed in
        </p>
        <button
          className="btn btn-success log-in"
          onClick={[Function]}
        >
          Log Out
        </button>
      </div>
    </div>
  </div>
</div>
`;
